src/simulation/LoadGenerator.cpp

```diff
@@ -458,43 +498,16 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
         }
     }
 
-    if (!mLoadTimer)
-    {
-        mLoadTimer = std::make_unique<VirtualTimer>(mApp.getClock());
-    }
-
     mStarted = true;
 
-    if (mApp.getState() == Application::APP_SYNCED_STATE)
-    {
-        mLoadTimer->expires_from_now(std::chrono::milliseconds(STEP_MSECS));
-        mLoadTimer->async_wait([this, cfg]() { this->generateLoad(cfg); },
-                               &VirtualTimer::onFailureNoop);
-    }
-    else
+    if (!mStartTime)
```

Comment:
I think we should change this to `releaseAssert(!mStartTime)` (assuming you implement my other comment where we exit early in `start()`).